### PR TITLE
Addition of an API for Beacons. BUKKIT-2842

### PR DESCRIPTION
**Pull request by deathmarine**

### CODE REVIEW

**Comment by turt2live** [comment]
Squash it to one commit (so you don't have to do it 6 months from now)

**Comment by deathmarine** [comment]
Wierd, I swear I thought I amended that commit.

File: src/main/java/org/bukkit/block/Beacon.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     * @return The primary level of the beacon.
+     */
+    public int getPrimary();
+
+    /**
+     * Sets the primary level for the beacon.
+     *
+     * @param level - The primary level of the beacon.
+     */
+	public void setPrimary(int level);
```
Comment (jrtc27) [inline src/main/java/org/bukkit/block/Beacon.java:25]: Indentation

File: src/main/java/org/bukkit/block/Beacon.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Represents a Beacon
+ */
+public interface Beacon extends BlockState{
+    /**
+     * Gets the primary level for the beacon.
+     *
+     * @return The primary level of the beacon.
+     */
+    public int getPrimary();
```
Comment (jrtc27) [inline src/main/java/org/bukkit/block/Beacon.java:18]: Indentation

**Comment by jrtc27** [comment]
Javadoc needs cleaning up - why is there a `-` after your parameters?

File: src/main/java/org/bukkit/event/block/BeaconPotionEffectEvent.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+     * @return Applied potions.
+     */
+    public Set<PotionEffect> getEffects() {
+        return this.potions;
+    }
+
+    /**
+     * Is the event cancelled?
+     * 
+     * @return Is it?
```
Comment (jrtc27) [inline src/main/java/org/bukkit/event/block/BeaconPotionEffectEvent.java:73]: Maybe change this and `setCancelled(boolean)` to inherit Javadoc from `Cancellable`?

File: src/main/java/org/bukkit/block/Beacon.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.Collection;
+import java.util.List;
+
+import org.bukkit.Material;
+import org.bukkit.potion.PotionEffect;
+
+/**
+ * Represents a Beacon
+ */
+public interface Beacon extends BlockState{
```
Comment (jrtc27) [inline src/main/java/org/bukkit/block/Beacon.java:12]: Should be a space between `BlockState` and `{`

**Comment by riking** [comment]
Open up Eclipse, open this file, hit `Ctrl-A` `Ctrl-Shift-F` (all-format), then double-check everything to make sure it's at the Bukkit coding standards.

**Comment by SagaciousZed** [comment]
The JavaDoc needs to be more consistent with itself. Additionally, if I can gain the same information by reading the method name, the JavaDoc needs to improve.

File: src/main/java/org/bukkit/event/block/BeaconPotionEffectEvent.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+     * Cancel or uncancel the event.
+     * 
+     * @param isCancelled New cancelled state.
+     */
+    public void setCancelled(boolean isCancelled) {
+        this.isCancelled = isCancelled;
+    }
+
+    /**
+     * Set affected players.
```
Comment (riking) [inline src/main/java/org/bukkit/event/block/BeaconPotionEffectEvent.java:35]: Set the list of players affected by this event.

**Comment by Pheenixm** [comment]
Has this been abandoned? It's a real shame, I need this

**Comment by turt2live** [comment]
@deathmarine If you are still interested in this you have plenty of work to do. This will need to be updated to the latest code base, edited to match the CONTRIBUTING.md guidelines, and updated to address the above concerns.

Once those three things are completed the review process can continue.

**Comment by turt2live** [comment]
I'm not reviewing it if it's not in a PR. Please also don't randomly ask for a review on your code on someone else's PR

**Comment by riking** [comment]
Okay. I deleted the comment to avoid polluting this discussion.

**Comment by turt2live** [comment]
Currently this PR is missing the following:
- Testing materials / Proof of concept (just to show that this indeed works)
- The ability to be pulled (currently the code is outdated where a merge is not possible. I would personally hold off on this until the 1.7 update is completed)
- The standards introduced after this PR's creation will need to be met

If anyone can tackle these tasks, the chance of this PR getting pulled significantly increases.

**Comment by Wolvereness** [comment]
PR abandoned.

**Pull request closed without merging**